pkg/server/share_manager.go

```diff
@@ -47,6 +60,11 @@ func NewShareManager(logger logrus.FieldLogger, volume volume.Volume) (*ShareMan
 	}
 	m.context, m.shutdown = context.WithCancel(context.Background())
 
+	err := m.makeLeaseClient()
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to make lease client - no lease renewal will be done")
+	}
```

Comment:
Why not handle the error?
If the lease setting is enabled, share manager needs to respect the it.

Reply:
Ok, I hear a requirement to make longhorn-share-manager sensitive to a setting value.  That's a first, I think, among non-manager pods, but not a problem.  It already has a kubernetes api client to get at leases, so not hard to make it look at settings, too.

Reply:
A share-manager pod can get the setting value either from an environment variable specified in the share-manager pod manifest or from the Longhorn client API when the pod starts.
Once the pod is running, it does not need to recheck the setting periodically. This means any changes to the setting will not be applied to the running share-manager pod. 
WDYT? @PhanLe1010 @james-munson

Reply:
@PhanLe1010 leans toward environment variable and avoiding API access where possible.  So that's what I've done.